src/sampling.py

```diff
@@ -22,6 +22,51 @@ def mnist_iid(dataset, num_users):
         all_idxs = list(set(all_idxs) - dict_users[i])
     return dict_users
 
+def split_dirichlet(dataset, num_users, is_cfar, beta=0.5):
```

Comment:
It would be nice if you can add dynamic typing for each parameter and return values, especially when it comes to numpy arrays where dimensions can vary